packages/concerto-ui-react/src/concertoForm.js

```diff
@@ -146,7 +134,7 @@ class ConcertoForm extends Component {
 
   addElement(e, key, value){
     const array = get(this.state.value, key);
-    set(this.state.value,`${key}.${array.length}`, value);
+    set(this.state.value, [...key, array.length], value);
```

Comment:
If I understand this correctly, this will update the state object without using `this.setState`, which _could_ cause issues in React. State should be considered immutable and only updated with `this.setState` in class components so that React recognizes the state change and re-renders accordingly.

I would suggest first doing a deep clone of the `value` object, updating that clone, and then using `this.setState` to update `value` on state with the new object. 

To note - there are a few other places in this file that also follow this same anti-pattern.

Reply:
Thanks Diana, fixed now